trieste/space.py

```diff
@@ -329,3 +347,153 @@ def __mul__(self, other: Box) -> Box:
 
     def __deepcopy__(self, memo: dict[int, object]) -> Box:
         return self
+
+
+class TaggedProductSearchSpace(SearchSpace):
+    r"""
+    Product :class:`SearchSpace` consisting of a product of
+    multiple :class:`SearchSpace`s. This class provides functionality for
```

Comment:
Apparently you can't attach an s to a reference like this.

```suggestion
    multiple :class:`SearchSpace`. This class provides functionality for
```